frontend/frontend.go

```diff
@@ -174,39 +174,72 @@ func (f *Frontend) HealthzReady(writer http.ResponseWriter, request *http.Reques
 
 func (f *Frontend) ArmResourceListBySubscription(writer http.ResponseWriter, request *http.Request) {
 	ctx := request.Context()
-	logger := ctx.Value(ContextKeyLogger).(*slog.Logger)
-	versionedInterface := ctx.Value(ContextKeyVersion).(api.Version)
 
-	logger.Info(fmt.Sprintf("%s: ArmResourceListBySubscription", versionedInterface))
+	versionedInterface, ok := VersionFromContext(ctx)
+	if !ok {
+		err := &ContextError{
+			got: versionedInterface,
+		}
```

Comment:
It looks like every place where we handle `!ok` we create a `ContextError`.  Can we have these `*FromContext` functions just return a `ContextError` instead of a `bool`?

Reply:
I thought about it but I was not sure if everywhere I was doing !ok I would create an error, but now I see this is the case, so done. Thanks! 🙂